swingset/src/main/java/com/nqadmin/swingset/SSComboBox.java

```diff
@@ -366,12 +382,22 @@ public void setModel(ComboBoxModel<SSListItem> model) {
 	/**
 	 * Convenience method for accessing the model with proper casting.
 	 * 
-	 * @return mapping list model with proper casting
+	 * @return mapping list model with proper casting, may be null
 	 */
 	protected OptionMappingSwingModel<Integer, String, Object> getOptionModel() {
 		return optionSwingModel;
 	}
 
+	@Override
+	public int getSelectedIndex() {
+		//if (ssCommon != null && getAllowNull()) {
+		//	throw new IllegalStateException("getAllowNull() true");
+		//}
+		return super.getSelectedIndex();
```

Comment:
This is a little tricky. My initial analysis was flawed, and JCombo uses this a lot. I'll check the stack to see if this is called from "outside". Does nqadmin, other than swingset, identify the app?

Reply:
com.nqadmin is in the package name for jdbcrowsetimpl and for our internal apps. Wondering if GlazedList calls it too?

Reply:
I had another thought on this, but seems dangerous especially if other stuff is relying on getSelectedIndex(). We could see if mappings are null (meaning 0 to n-1 mappings) **and** getAllowNull **and** if super.getSelectedIndex() >0 then return getSelectedIndex-1. It was a bad idea in my head and worse now that I've actually typed it out. Ugh.

Reply:
I may just have to suck it up and fix our code to not use getSelectedIndex(). As an intermediate measure we could add a new temporary method such as getSelectedIndexNoFirstNull() that did some adjustment as outline above: if getAllowNull()==true then return -1 for -1 or 0, otherwise return getSelectedIndex()-1.

Reply:
I've got everything working last night, doing final reviews, may not have time to push until this afternoon (afternoon here). Use of getSelectedIndex throws as discussed.

I'm confused about what you're looking for. I though the issue with getSelectedIndex was for [0-N) mappings. Where you wanted to get the mapping by using the index. So
```
Integer iInt = getSelectedMapping();
int indexIsMapping = iInt != null ? iInt : -1;
```
This doesn't work if it's not a [0,N) mappings.

In any case, feel free to add any transitional methods you need. I wouldn't (and can't) stop you

Reply:
Perhaps the simplest, would be to put getAllowNull(false) in the constructor, stuff should work as is. Then if you want to use whatever the metadata says, the caller could do combo.setAllowNull(null) which means use the metadata. Or use combo.setAlowNull(true) to put the nullItem in no matter what the metadata says.

Reply:
```
boolean noSelection = combo.getSelectedMapping() == null;
```
Is equivelent to
```
boolean noSelection = combo.getSelectedIndex() == -1;
```
in the pre nullItem days.